linode/manager
# refactor: [M3-8770] - Clean up Profile Display Settings page

## Description 📝

- Does some clean up of the http://localhost:3000/profile/display page 🧹 

## Changes  🔄
- Uses `<form>`s so that the user can press `Enter` to submit forms
- Removes the `SingleTextFieldForm.tsx` component
  - The abstraction wasn't widely adopted
  - Allows us to clean up a use of the `uuid` package (one less file to update as part of pnpm)
- Uses a Toast notification for all success states rather than a success notice 🍞 
  - Currently in develop, the timezone uses a success toast but email and username use a notice
- Breaks up `DisplaySettings` into more components 🧩
- Improves unit testing 🧪 
- Uses `useUpdateUserMutation` rather than `updateUser` directly from api-v4
  - This improves a small flicker on the "Save Username" button and helps keep cache up to date

## Preview 📷

|  Before  | This PR   |
| ------- | ------- |
| ![Screenshot 2024-10-22 at 3 56 30 PM](https://github.com/user-attachments/assets/fb739aa3-db1f-4d71-a4ed-014dbc87a0d1) | ![Screenshot 2024-10-22 at 3 55 42 PM](https://github.com/user-attachments/assets/c597edb3-123c-4e73-b24a-5a1ec47b4ca6) |
| ![Screenshot 2024-10-22 at 4 12 58 PM](https://github.com/user-attachments/assets/ba7f5fb3-3f31-4ea2-9616-da929ac19bde) | ![Screenshot 2024-10-22 at 4 12 45 PM](https://github.com/user-attachments/assets/cccdd716-2a36-4150-925f-5dbc4227d2c8) |

## How to test 🧪

- Test general functionality of the http://localhost:3000/profile/display page 👀 
- Verify this PR's changes do not result in any new behavior - general functionality should be the same
- Check context and quality of new unit tests 🧪 

## As an Author I have considered 🤔
